src/app/pages/wallets/my-wallets/pages-wallets-my-wallets.component.ts

```diff
@@ -89,4 +96,26 @@ export class PagesWalletsMyWalletsComponent implements OnInit {
 
     return dialogRef.afterClosed();
   }
+
+  public handleWalletDelete(wallet: MyWallet): void {
+
+    this.confirmDialogService.openConfirmModal({
+      headerText: `Deleting ${wallet.name} wallet`,
+      confirmationText: `Are you sure you want to delete ${wallet.name} wallet and all related data?`,
+    }).subscribe((result: boolean | undefined) => {
+      if (!result) {
+        return;
+      }
+      this.deleteWallet(wallet);
+
+    });
+  }
+
+  private deleteWallet(wallet: MyWallet): void {
+    this.loadingDialogService.show('Deleting wallet')
+    this.myWalletsService.deleteWallet(wallet).subscribe(() => {
+      this.loadingDialogService.hide();
```

Comment:
You should also remember about hiding loading dialog whenever there is an error - so not only in success scenario.